nautobot/core/tests/test_tables.py

```diff
@@ -63,3 +67,55 @@ def test_tree_model_table_orderable(self):
         queryset = RackGroupTable.Meta.model.objects.annotate(rack_count=count_related(Rack, "rack_group")).all()
         self._validate_sorted_tree_queryset_same_with_table_queryset(queryset, RackGroupTable, "rack_count")
         self._validate_sorted_tree_queryset_same_with_table_queryset(queryset, RackGroupTable, "-rack_count")
+
+    def test_base_table_apis(self):
+        """
+        Test BaseTable APIs, specifically visible_columns and configurable_columns.
+        Assert that they gave the correct results.
+        """
+        wn_table = WirelessNetworkTable(WirelessNetwork.objects.all())
+        location_table = LocationTable(Location.objects.all())
+        circuit_table = CircuitTable(Circuit.objects.all())
+
+        # Wireless Network Table
+        for name, column in wn_table.columns.items():
+            expected_visible_columns = [
+                name for name, column in wn_table.columns.items() if column.visible and name not in wn_table.exclude
+            ]
+            expected_configurable_columns = [
+                (name, column.verbose_name)
+                for name, column in wn_table.columns.items()
+                if name not in ["pk", "actions"]
+            ]
```

Comment:
Feels a bit cheaty to just copy/paste the same logic into the test case. :-)

Can we instead actually mess with the tables a bit (`LocationTable(Location.objects.all(), exclude=[...])`, `location_table.columns["foo"].hide()`, etc.) and assert some specific lists of expected values from these APIs as a consequence?

Reply:
That's what I did originally, my concern is that we need to change the test case every time we add/remove columns from the table. But I can change it.